data/mods/ssb/abilities.ts

```diff
@@ -279,6 +279,18 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
 			pokemon.formeChange('Shaymin-Sky', this.effect);
 		},
 	},
+	// Celine
+	guardianarmor: {
+		shortDesc: "Raises Defense and Special Defense by two stages upon switch in.",
+		isNonstandard: "Custom",
+		onModifyDef(def) {
+			return this.chainModify(2);
+		},
+		onModifySpD(spd) {
+			return this.chainModify(2);
+		},
```

Comment:
This is just a direct stat modifier (like Eviolite or Fur Coat) so I would look at Drampa's abilities for a better idea on how to do this